packages/dashboard/src/contexts/DashContext/state.ts

```diff
@@ -0,0 +1,122 @@
+import type { providers } from "ethers";
+import type {
+  DashboardMessageBusClient,
+  ReceivedMessageLifecycle
+} from "@truffle/dashboard-message-bus-client";
+import {
+  isDashboardProviderMessage,
+  isInvalidateMessage,
+  isLogMessage,
+  isDebugMessage
+} from "@truffle/dashboard-message-bus-common";
+import type {
+  Message,
+  DashboardProviderMessage
+} from "@truffle/dashboard-message-bus-common";
+import {
+  messageNeedsInteraction,
+  messageIsUnsupported,
+  rejectMessage,
+  confirmMessage
+} from "src/utils/dash";
+import type noticeContentType from "src/components/composed/Notice/noticeContentType";
+
+type actionType =
+  | { type: "set-client"; data: DashboardMessageBusClient }
+  | { type: "set-chain-info"; data: stateType["chainInfo"] }
+  | { type: "set-notice"; data: Partial<stateType["notice"]> }
+  | {
+      type: "handle-message";
+      data: {
+        lifecycle: ReceivedMessageLifecycle<Message>;
+        provider: providers.JsonRpcProvider;
+      };
+    };
+
+type stateType = {
```

Comment:
(same here with the capitalization)

Also perhaps worth noting... it's better to use `interface` when you can, since it's a simpler concept... sometimes you can't, but here I'm sure `interface` is fine